packages/devtools_app/lib/src/error_badge_manager.dart

```diff
@@ -43,16 +44,52 @@ class ErrorBadgeManager extends DisposableController
     if (e.extensionKind == 'Flutter.Error') {
       incrementBadgeCount(LoggingScreen.id);
 
-      final json = e.extensionData.data;
-      final objectId = json['objectId'] as String;
-      // TODO(kenz): verify that the object id is referring to an element or
-      // widget.
-      if (objectId?.contains('inspector-') ?? false) {
-        incrementBadgeCount(InspectorScreen.id);
+      final inspectableError = _extractInspectableError(e);
+      if (inspectableError != null) {
+        appendInspectorErroredWidget(inspectableError);
       }
     }
   }
 
+  InspectableWidgetError _extractInspectableError(Event error) {
+    final json = error.extensionData.data;
+
+    // TODO(dantup): Is there a better way to get the inspectorRef we need?
+    final properties = json['properties'] as List<dynamic>;
+
+    final errorSummaryNode = properties
+        ?.firstWhere((p) => p['type'] == 'ErrorSummary', orElse: () => null);
+    final errorMessage = errorSummaryNode != null
+        ? errorSummaryNode['description'] as String
+        : null;
+    if (errorMessage == null) {
+      return null;
+    }
+
+    final devToolsUrlNode = properties?.firstWhere(
+        (p) => p['type'] == 'DevToolsDeepLinkProperty' && p['value'] != null,
+        orElse: () => null);
+    if (devToolsUrlNode == null) {
+      return null;
+    }
+
+    var inspectWidgetUrl = Uri.tryParse(devToolsUrlNode['value'] as String);
+    if (inspectWidgetUrl == null) {
+      return null;
+    }
+
+    // Handle when querystring is in the fragement.
```

Comment:
See `devToolsQueryParams` in utils.dart - you may be able to use this here.

Reply:
Thanks! Done in 6f779c11